### PR TITLE
Two batches of fixes to make productivity workloads run on more devices

### PR DESCRIPTION
**Pull request by jimboatarm**

### CODE REVIEW

File: wlauto/workloads/gmail/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java [hunk 1]
```diff
@@ -97,6 +100,10 @@ public void clickSendButton() throws Exception {
         clickUiObject(sendButton, timeout);
         result.end();
         timingResults.put("Send", result);
+
+        UiObject drawerButton = new UiObject(new UiSelector().descriptionContains("avigat")
```
Comment (muendelezaji) [inline wlauto/workloads/gmail/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java:104]: It is not clear what "avigat" is meant to find, presumably a variant of 'Navigation' ?

File: wlauto/workloads/gmail/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java [hunk 2]
```diff
@@ -17,7 +17,8 @@
     public static String TAG = "uxperf_gmail";
 
     private Bundle parameters;
-    private long networkTimeout =  TimeUnit.SECONDS.toMillis(20);
+    private int networkTimeoutSecs = 20;
+    private long networkTimeout =  TimeUnit.SECONDS.toMillis(networkTimeoutSecs);
```
Comment (muendelezaji) [inline wlauto/workloads/gmail/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java:21]: unnecessary change as it's already self-descriptive? niggle alert! :)

File: wlauto/workloads/gmail/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java [hunk 3]
```diff
@@ -61,7 +62,7 @@ public void clickNewMail() throws Exception {
 
     public void setToField() throws Exception {
         Timer result = new Timer();
-        UiObject toField = getUiObjectByDescription("To", "android.widget.TextView");
+        UiObject toField = getUiObjectByText("To", "android.widget.TextView");
```
Comment (muendelezaji) [inline wlauto/workloads/gmail/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java:65]: Is it worth searching by both text & description or has this proven to work everywhere?

File: wlauto/workloads/gmail/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java (hunk 2 above)
Comment (jimboatarm) [inline wlauto/workloads/gmail/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java:21]: Look down a bit, I use both the milliseconds and second variants later on.

File: wlauto/workloads/gmail/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java (hunk 3 above)
Comment (jimboatarm) [inline wlauto/workloads/gmail/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java:65]: This is the one that works everywhere.   On some devices the description is empty - yes, really!

File: wlauto/workloads/gmail/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java (hunk 1 above)
Comment (jimboatarm) [inline wlauto/workloads/gmail/uiauto/src/com/arm/wlauto/uiauto/UiAutomation.java:104]: Good point.  I need to explain why this is done this way.

**Comment by muendelezaji** [comment]
Ok with this.
